packages/react/src/components/Authenticator/shared/__tests__/FormFields.test.tsx

```diff
@@ -0,0 +1,76 @@
+import { render, screen } from '@testing-library/react';
+import * as React from 'react';
+
+import { defaultFormFieldOptions } from '@aws-amplify/ui';
+import { FormField, FormFieldProps } from '../FormField';
+import * as UseAuthenticatorModule from '../../hooks/useAuthenticator';
+import * as UseStableIdModule from '../../../../primitives/utils/useStableId';
+
+const usernameFormFieldProps: FormFieldProps = {
+  ...defaultFormFieldOptions.username,
+  name: 'username',
+  label: 'Username',
+};
+
+const useAuthenticatorSpy = jest.spyOn(
+  UseAuthenticatorModule,
+  'useAuthenticator'
+);
+
+const errorMessage = 'Username cannot be empty.';
+
+const mockValidFormField = () => {
+  useAuthenticatorSpy.mockReturnValue({ validationErrors: {} } as any);
+};
+
+const mockInvalidFormField = () => {
+  useAuthenticatorSpy.mockReturnValue({
+    validationErrors: { username: errorMessage },
+  } as any);
+};
+
+const useStableIdSpy = jest.spyOn(UseStableIdModule, 'useStableId');
+
+describe('FormField Snapshots: ', () => {
+  beforeAll(() => {
+    // mocking ids so snapshots are consistent
+    useStableIdSpy.mockReturnValue('mock-id-0');
```

Comment:
84fd456, I used `mockReturnValue` followed with `.mockRestore`, because this hooks are called more than once on re-renders and we aren't really concerned with the exact number it gets called.